Bottom: c9274c595cf5ae68ca692820deb69760b7cd1461
Top:    206e1412f7d5b99ecf17a9f625fc95924d650a15
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-02 14:16:11 +0000

Refresh of push-special-case-thindraw-vs-

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 04f1e05..bdd2e9b 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -603,6 +603,9 @@ ghid_fill_rect (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2)
 static void
 ghid_draw_pcb_polygon (hidGC gc, PolygonType *polygon, const BoxType *drawn_area)
 {
+  if (polygon->Clipped == NULL)
+    return;
+
   USE_GC (gc);
 
   if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
